hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -205,10 +214,11 @@ def print_end(self):
         By default does nothing.
         """
 
-    def print_step(self, step):
+    def print_step(self, step, result=None):
```

Comment:
I am unsure if `print_step` is part of the API for this class. If it is not, then we don't need a default value for the `result` parameter

Reply:
It's [not part of the public API](https://hypothesis.readthedocs.io/en/latest/stateful.html#generic-state-machines), so having a default or not are both fine.

(we'll be refactoring it all after dropping GenericStateMachine either way)